### PR TITLE
Add 2D shadows and canvas SDF to OpenGL3 renderer

### PR DESCRIPTION
**Pull request by clayjohn**
This is an initial implementation based on the current RD implementation. I have plans to return to these later and see if we can improve performance at the expense of some quality. For shadows, performance is fine, but could be better. For SDF, performance is not amazing unless running on a dedicated GPU, I would like to make SDF more feasible on mobile and integrated GPUs, but that will have to wait as it is an advanced feature anyway.

Screenshot from 2022-10-18 21-21-54

Tested on Desktop and on Web

### CODE REVIEW

File: drivers/gles3/storage/texture_storage.cpp [hunk 1]
```diff
@@ -1755,13 +1713,278 @@ void TextureStorage::render_target_do_clear_request(RID p_render_target) {
...
+		return texture->tex_id;
+	}
+
+	return rt->sdf_texture_read;
+}
+
+void TextureStorage::_render_target_allocate_sdf(RenderTarget *rt) {
+	ERR_FAIL_COND(rt->sdf_texture_write_fb != 0);
+	if (rt->sdf_texture_read != 0) {
+		glDeleteTextures(1, &rt->sdf_texture_read);
```
Comment (BastiaanOlij) [inline drivers/gles3/storage/texture_storage.cpp:1792]: Shouldn't we be checking all textures that we're about to create here? This code suggest we could have a situation where `_render_target_allocate_sdf` can be called without first calling clear and we would be leaking textures as a result.

Or is there some other reason we only check the `sdf_texture_read` texture and not the rest?

File: drivers/gles3/storage/texture_storage.cpp [hunk 2]
```diff
@@ -1755,13 +1713,278 @@ void TextureStorage::render_target_do_clear_request(RID p_render_target) {
...
+}
+
+void TextureStorage::_render_target_clear_sdf(RenderTarget *rt) {
+	if (rt->sdf_texture_read != 0) {
+		glDeleteTextures(1, &rt->sdf_texture_read);
+		rt->sdf_texture_read = 0;
+	}
+	if (rt->sdf_texture_write_fb != 0) {
+		glDeleteTextures(1, &rt->sdf_texture_write);
+		glDeleteTextures(1, &rt->sdf_texture_process[0]);
```
Comment (BastiaanOlij) [inline drivers/gles3/storage/texture_storage.cpp:1870]: why not `glDeleteTextures(2, rt->sdf_texture_process);`?

Comment (clayjohn) [reply drivers/gles3/storage/texture_storage.cpp:1870]: No reason, I'll change it

**Comment by clayjohn** [comment]
@BastiaanOlij Updated based on your comments! This should be good to go now!

File: drivers/gles3/storage/texture_storage.cpp (hunk 2 above)
Comment (BastiaanOlij) [reply drivers/gles3/storage/texture_storage.cpp]: Yeah nitpicking to the nth degree on my part, I'm a little OCD to match allocation and deallocation :)

**Review by BastiaanOlij — APPROVED** [review]

**Comment by BastiaanOlij** [comment]
I don't see anything obvious standing out, looks like a solid implementation. Can't judge it on how it actually works, still need to read up on SDF one of these days :)

**Comment by clayjohn** [comment]
Rebased, should be ready to merge :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**